src/Features/Admin/Categories/index.js

```diff
@@ -21,7 +22,24 @@ function CategoryList() {
     row;
   };
 
-  const header = ['No', 'Category', 'Edit'];
+  const [categories, setCategories] = useState([]);
+
+  const getCategoriesResponseFromEpic = (response) => {
+    setCategories(response);
+  };
+
+  const [header, setHeader] = useState([]);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchCategories(getCategoriesResponseFromEpic));
+  }, []);
+
+  useEffect(() => {
+    if (categories.length > 0) {
+      setHeader([...Object.keys(categories[0]), 'Edit']);
```

Comment:
Why we need to get the headers value based on response from API may be API have some extra payload such as createdDate and other things and if you use API payload then headers values will be way more than we expected